samza-core/src/main/java/org/apache/samza/config/JobCoordinatorConfig.java

```diff
@@ -22,33 +22,42 @@
 import com.google.common.base.Strings;
 import org.apache.samza.SamzaException;
 import org.apache.samza.coordinator.CoordinationUtilsFactory;
+import org.apache.samza.standalone.PassthroughCoordinationUtilsFactory;
+import org.apache.samza.standalone.PassthroughJobCoordinatorFactory;
 import org.apache.samza.util.Util;
 import org.apache.samza.zk.ZkCoordinationUtilsFactory;
+import org.apache.samza.zk.ZkJobCoordinatorFactory;
 
 public class JobCoordinatorConfig extends MapConfig {
   public static final String JOB_COORDINATOR_FACTORY = "job.coordinator.factory";
-  public static final String JOB_COORDINATION_UTILS_FACTORY = "job.coordination.utils.factory";
-  public final static String DEFAULT_COORDINATION_UTILS_FACTORY = ZkCoordinationUtilsFactory.class.getName();
+  public final static String DEFAULT_COORDINATOR_FACTORY = ZkJobCoordinatorFactory.class.getName();
+  private static final String AZURE_COORDINATION_UTILS_FACTORY = "org.apache.samza.coordinator.AzureCoordinationUtilsFactory";
+  private static final String AZURE_COORDINATOR_FACTORY = "org.apache.samza.coordinator.AzureJobCoordinatorFactory";
 
   public JobCoordinatorConfig(Config config) {
     super(config);
   }
 
   public String getJobCoordinationUtilsFactoryClassName() {
-    String className = get(JOB_COORDINATION_UTILS_FACTORY, DEFAULT_COORDINATION_UTILS_FACTORY);
+    String coordinatorFactory = get(JOB_COORDINATOR_FACTORY, DEFAULT_COORDINATOR_FACTORY);
 
-    if (Strings.isNullOrEmpty(className)) {
-      throw new SamzaException("Empty config for " + JOB_COORDINATION_UTILS_FACTORY + " = " + className);
+    String coordinationUtilsFactory;
+    if (AZURE_COORDINATOR_FACTORY.equals(coordinatorFactory)) {
+      coordinationUtilsFactory = AZURE_COORDINATION_UTILS_FACTORY;
+    } else if (PassthroughJobCoordinatorFactory.class.getName().equals(coordinatorFactory)) {
+      coordinationUtilsFactory = PassthroughCoordinationUtilsFactory.class.getName();
+    } else {
```

Comment:
By default, we intend to choose `ZkJobCoordinatorFactory`. So if the cooridnator factory is not overridden to `AzureJobCoordinatorFactory` or `PassThroughCoordinatory`, the right behavior should be set coordinatorUtils to `ZkJobCoordinatorUtilsfactory`. We should not throw up here.

Reply:
Please don't early-resolve discussions. I meant that you should throw a meaningful exception if the job coordinator factory is not one of the three listed here, so that developers know to add their special case here.

Reply:
Alright, will change it.